sonic_platform_base/sensor_base.py

```diff
@@ -0,0 +1,172 @@
+
+        Returns:
+            Sensor measurement unit
+        """
+        raise NotImplementedError
+
+    def get_high_threshold(self):
+        """
+        Retrieves the high threshold of sensor
+
+        Returns:
+            High threshold 
+        """
+        raise NotImplementedError
+
+    def get_low_threshold(self):
+        """
+        Retrieves the low threshold 
+
+        Returns:
+            Low threshold 
+        """
+        raise NotImplementedError
+
+    def set_high_threshold(self, value):
+        """
+        Sets the high threshold value of sensor
+
+        Args :
+            value: High threshold value to set
+
+        Returns:
+            A boolean, True if threshold is set successfully, False if not
+        """
+        raise NotImplementedError
+
+    def set_low_threshold(self, value):
+        """
+        Sets the low threshold value of sensor
+
+        Args :
+            value: Value 
+
+        Returns:
+            A boolean, True if threshold is set successfully, False if not
+        """
+        raise NotImplementedError
+
+    def get_high_critical_threshold(self):
+        """
+        Retrieves the high critical threshold value of sensor
+
+        Returns:
+            The high critical threshold value of sensor 
+        """
+        raise NotImplementedError
+
+    def get_low_critical_threshold(self):
+        """
+        Retrieves the low critical threshold value of sensor
+
+        Returns:
+            The low critical threshold value of sensor 
+        """
+        raise NotImplementedError
+
+    def set_high_critical_threshold(self, value):
+        """
+        Sets the critical high threshold value of sensor
+
+        Args :
+            value: Critical high threshold Value 
+
+        Returns:
+            A boolean, True if threshold is set successfully, False if not
+        """
+        raise NotImplementedError
+
+    def set_low_critical_threshold(self, value):
+        """
+        Sets the critical low threshold value of sensor
+
+        Args :
+            value: Critial low threshold Value 
+
+        Returns:
+            A boolean, True if threshold is set successfully, False if not
+        """
+        raise NotImplementedError
+
+    def get_minimum_recorded(self):
+        """
+        Retrieves the minimum recorded value of sensor
+
+        Returns:
+            The minimum recorded value of sensor 
+        """
+        raise NotImplementedError
+
+    def get_maximum_recorded(self):
+        """
+        Retrieves the maximum recorded value of sensor
+
+        Returns:
+            The maximum recorded value of sensor 
+        """
+        raise NotImplementedError
+
+
+
+class VsensorBase(SensorBase):
+    """
+    Abstract base class for interfacing with a voltage sensor module
+    """
+    MILLI_VOLTS = "mV"
+
+    def get_type(self):
+        return SENSOR_TYPE_VOLTAGE
+
+    def get_unit(self):
+        return MILLI_VOLTS
+
```

Comment:
Similar to ThermalBase with `get_temperature`, should there be APIs called `get_voltage` and `get_current`? They can just call `get_value`? I prefer it without these (like you currently have), making it more generic, but I'd be curious to know the community's opinion on this.

Reply:
sure. We can check in community review. I preferred this for commonality and brevity as u mentioned.